telegram/ext/builders.py

```diff
@@ -0,0 +1,1338 @@
+
+    def token(self: BuilderType, token: str) -> BuilderType:
+        """Sets the token to be used for :attr:`telegram.ext.Dispatcher.bot`.
+
+        Args:
+            token (:obj:`str`): The token.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_token(token)
+
+    def base_url(self: BuilderType, base_url: str) -> BuilderType:
+        """Sets the base URL to be used for :attr:`telegram.ext.Dispatcher.bot`. If not called,
+        will default to ``'https://api.telegram.org/bot'``.
+
+        .. seealso:: :attr:`telegram.Bot.base_url`, `Local Bot API Server <https://github.com/\
+            python-telegram-bot/python-telegram-bot/wiki/Local-Bot-API-Server>`_,
+            :meth:`base_url`
+
+        Args:
+            base_url (:obj:`str`): The URL.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_base_url(base_url)
+
+    def base_file_url(self: BuilderType, base_file_url: str) -> BuilderType:
+        """Sets the base file URL to be used for :attr:`telegram.ext.Dispatcher.bot`. If not
+        called, will default to ``'https://api.telegram.org/file/bot'``.
+
+        .. seealso:: :attr:`telegram.Bot.base_file_url`, `Local Bot API Server <https://github.com\
+            /python-telegram-bot/python-telegram-bot/wiki/Local-Bot-API-Server>`_,
+            :meth:`base_file_url`
+
+        Args:
+            base_file_url (:obj:`str`): The URL.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_base_file_url(base_file_url)
+
+    def request_kwargs(self: BuilderType, request_kwargs: Dict[str, Any]) -> BuilderType:
+        """Sets keyword arguments that will be passed to the :class:`telegram.utils.Request` object
+        that is created when :attr:`telegram.ext.Dispatcher.bot` is created. If not called, no
+        keyword arguments will be passed.
+
+        .. seealso:: :meth:`request`
+
+        Args:
+            request_kwargs (Dict[:obj:`str`, :obj:`object`]): The keyword arguments.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_request_kwargs(request_kwargs)
+
+    def request(self: BuilderType, request: Request) -> BuilderType:
+        """Sets a :class:`telegram.utils.Request` object to be used for
+        :attr:`telegram.ext.Dispatcher.bot`.
+
+        .. seealso:: :meth:`request_kwargs`
+
+        Args:
+            request (:class:`telegram.utils.Request`): The request object.
+
+        Returns:
+            :class:`DispatcherBuilder`: The same builder with the updated argument.
+        """
+        return self._set_request(request)
+
+    def private_key(self: BuilderType, private_key: bytes) -> BuilderType:
```

Comment:
I wonder if `private_key` and `private_key_password` should be merged into a single setter.
After all, these two arguments will always be used together, they are checked together, etc. etc. (we might also want to fix the internal implementation as well).

In a sense, this is similar to how we allow setting a custom Dispatcher class, and taking kwargs as well.

Reply:
that makes sense 👍